tests/test_server.py

```diff
@@ -95,6 +95,42 @@ def test_get_pet_not_found(self):
     resp = self.app.get('/inventory/products/0')
     self.assertEqual(resp.status_code, server.HTTP_404_NOT_FOUND)
 
+  def test_update_product(self):
```

Comment:
All these should be split into different functions I think

- test_update_product_with_exceding_quantity()
- test_update_product_with_negative_quantity()